src/lib/util/Config.ts

```diff
@@ -215,31 +183,36 @@ export class Config {
-		return this.getConfigValue<boolean>(propertyName, engine, this.typeChecker.booleanCheck);
+		return this.getConfigValue<boolean>(propertyName, engine, booleanTypeGuard, 'InvalidBooleanValue');
 	}
 
 	protected getNumberConfigValue(propertyName: string, engine: ENGINE): Promise<number> {
-		return this.getConfigValue<number>(propertyName, engine, this.typeChecker.numberCheck);
+		return this.getConfigValue<number>(propertyName, engine, numberTypeGuard, 'InvalidNumberValue');
 	}
 
 	protected getStringArrayConfigValue(propertyName: string, engine: ENGINE): Promise<string[]> {
-		return this.getConfigValue<string[]>(propertyName, engine, this.typeChecker.stringArrayCheck);
+		return this.getConfigValue<string[]>(propertyName, engine, stringArrayTypeGuard, 'InvalidStringArrayValue');
 	}
 
-	private async getConfigValue<T>(propertyName: string, engine: ENGINE, typeChecker: (any, string, ENGINE) => boolean): Promise<T> {
+	private async getConfigValue<T>(propertyName: string, engine: ENGINE, typeChecker: GenericTypeGuard<T>, errTemplate: string): Promise<T> {
 		let ecc = this.getEngineConfig(engine);
-		// If the config specifies property, use those.
-		// Intentionally distinguishing null from undefined in case we want to allow nullable config values in the future.
-		// TODO: Look into possibly using typeguards, see discussion here https://github.com/forcedotcom/sfdx-scanner/pull/222/files#r497080143
-		if (ecc && (ecc[propertyName] !== undefined) && typeChecker(ecc[propertyName], propertyName, engine)) {
-			this.logger.trace(`Retrieving ${propertyName} from engine ${engine}: ${ecc[propertyName]}`);
-		} else {
-			// If the config doesn't specify the property, use the default value
-			this.logger.trace(`Looking for missing property ${propertyName} of ${engine} in defaults`);
+		// If the config for this engine is undefined or the requested property is undefined, update to the default value
+		// and persist it for future use. Note: We're intentionally distinguishing between null and undefined, in case
+		// we want to allow nullable config properties in the future.
+		if (!ecc || ecc[propertyName] === undefined) {
+			this.logger.trace(`Looking for missing config property ${propertyName} of engine ${engine} in default configs`);
 			ecc = await this.lookupAndUpdateToDefault(engine, ecc, propertyName);
+		} else {
+			this.logger.trace(`Retrieving property ${propertyName} from existing config for engine ${engine}`);
+		}
+		// At this point, the config definitely has a value, so we can typecheck it and return it.
+		const propertyValue = ecc[propertyName] as unknown;
+		if (typeChecker(propertyValue)) {
+			this.logger.trace(`Config property ${propertyName} for engine ${engine} has value ${String(propertyValue)}`);
+			return propertyValue;
+		} else {
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'Config', errTemplate, [propertyName, engine.valueOf(), String(propertyValue)]);
```

Comment:
Instead of each typechecker throwing its own different type of error, the method throws the provided error message if the provided type checker fails. The alternative would have been to make the new utility functions in `Utils` throw errors (thereby reducing their utility) or to wrap them in other methods (too messy, imo).